### DISCUSSION TITLE
NordicTrack RX800 Default Settings

### DISCUSSION THREAD

**Discussion by alistairdickie** [JaapvanEkris/openrowingmonitor#58] [Rower Profile Development]
I'm finding it tricky to get the config right for a new NordicTrack RX800 that I inherited from a friend (without a monitor, hence getting ORM set up with a fresh install on a brand new Pi Zero 2 W). Any tips?

I have the damper set to 6 because it feels pretty much like a drag factor on the Concept 2 I am very familiar with for a 130 drag factor on the PM5. The default settings did not seem to be right for stroke detection so I have altered it a bit. The current config works ok if I row slow, but as soon as I power up it seems to be missing strokes. I have attached the raw log, rowing data and config file. The config was copied and modified from rowerProfiles.js. 
[rowingData.zip](https://github.com/user-attachments/files/17440553/rowingData.zip)

I checked to make sure I had four pulses per revolution of the flywheel. The raw data does not look noisy. 

This graph shows the timings of the detected strokes (from rowingdata.csv, plotted in red at 0.04 just to show them next to the peaks) overlayed on top of the raw data. At about 35 seconds I power up a bit and the strokes are not being recognised. When I power off again slightly at 60 seconds the detection starts to work again. 

![Untitled](https://github.com/user-attachments/assets/42876a9c-26ca-4c9f-ae43-c4ab88f2a65d)

Key settings are: 
    numOfImpulsesPerRevolution: 4,
    minimumTimeBetweenImpulses: 0.015,
    maximumTimeBetweenImpulses: 0.06,
    sprocketRadius: 3.0,
    autoAdjustDragFactor: true,
    minimumDragQuality: 0.83,
    dragFactorSmoothing: 3,
    flywheelInertia: 0.189,
    dragFactor: 460,
    flankLength: 9,
    minimumStrokeQuality: 0.34,
    minumumRecoverySlope: 0.001,
    autoAdjustRecoverySlope: true,
    autoAdjustRecoverySlopeMargin: 0.036,
    minumumForceBeforeStroke: 20,
    minimumDriveTime: 0.28,
    minimumRecoveryTime: 0.90

**Comment by JaapvanEkris** [comment]
Congrats on getting a nice machine (used to own one myself). I can imagine the settings might need an update (last real test was over two years ago, and the engine looked different then).

What version are you running, main or architecture_update?

Is there a specicific reason you changed the following settings:
* flywheelInertia
* flankLength
* minumumForceBeforeStroke

Especially flankLength could create a lot of noise that hinders stroke detection.

**Reply by JaapvanEkris** [reply]
Try these settings, they work on the latest version (i.e. Architecture_Updates):
```
  // NordicTrack RX800 Air Rower
  NordicTrack_RX800: {
    numOfImpulsesPerRevolution: 4,
    minimumTimeBetweenImpulses: 0.005,
    maximumTimeBetweenImpulses: 0.045,
    sprocketRadius: 3.0,
    autoAdjustDragFactor: true,
    minimumDragQuality: 0.83,
    dragFactorSmoothing: 3,
    flywheelInertia: 0.180,
    dragFactor: 225,
    flankLength: 11,
    minimumStrokeQuality: 0.34,
    minimumRecoverySlope: 0.001,
    autoAdjustRecoverySlope: true,
    autoAdjustRecoverySlopeMargin: 0.1,
    minimumForceBeforeStroke: 80,
    minimumDriveTime: 0.30,
    minimumRecoveryTime: 0.60
  },
```
The dragfactor is doing weird things: it hits 850 or above, which isn't normal for this machine (but with a Goodness Of Fit well over 99%, it isn't the data causing issues). Might be that the magnet is stuck somewhere at the far end. As a result, you need to change autoAdjustRecoverySlopeMargin as the dragfactor seems to drop significantly as you pick up speed.

**Comment by alistairdickie** [comment]
I had followed the default install and that got me the main branch. With a little thinking I managed to work out I could update to the most recent development branch using the command:

`updateopenrowingmonitor.sh -b V1Beta-Architecture_Revision`

Note that there were a few errors and warnings, mostly due to dependencies.

I also updated the config to the settings you list above. It is much better, but I am still getting missed strokes. What do you mean by the magnet getting stuck at the far end?
[2024-10-19_21-01-00_raw.csv.gz](https://github.com/user-attachments/files/17447383/2024-10-19_21-01-00_raw.csv.gz)
[2024-10-19_21-01-00_rowingData.csv](https://github.com/user-attachments/files/17447384/2024-10-19_21-01-00_rowingData.csv)

**Reply by Abasz** [reply]
> 850 or above, which isn't normal for this machine (but with a Goodness Of Fit well over 99%, it isn't the data causing issues).

I have these drag factor drops on the very dirty and old kayak first ergometers and I tracked it down to bearing and other mechanical drag. Their drag effect (since that is constant) decrease exponentially when flywheel speeds up. In these ergs the difference could have 20% and its very tricky to calibrate in settings well. So I set the stroke detection to pure slope based (i.e. disable the reliance on drag factor for the stroke detection). 

But first I would check whether the flywheel spins free properly and that is has properly lubricated.

**Reply by JaapvanEkris** [reply]
> It is much better, but I am still getting missed strokes. What do you mean by the magnet getting stuck at the far end?

The RX800 is a hybrid rower: it combines two flywheels (!) with a magnet breaking system. If you look at https://www.manualslib.com/manual/3144208/Nordictrack-Rx-800.html?page=19#manual and look for part 92, that is the resistance magnet, that is controlled by a rod connected to the resistance handle. If it is stuck in the far end, it will provide maximum magnetic resistance. The effect is similar to the effect described by @Abasz: the resistance is constant (i.e. speed independent) and thus its relative effect drops when the fan starts doing its (speed dependent) work at higher speeds.

**Reply by alistairdickie** [reply]
I have checked the physical condition of the machine. It is in near new hardly used condition. The flywheels spin freely. I took it apart and the magnet moves as expected (at 10 it is providing some resistance, at 2 it is not). I do not think that it is the mechanical condition that is the cause of the issue. 

With the drag set at 10 on the rower OPM seems to pick up every stroke and work pretty much as I would expect. See this data:
[2024-10-21_07-48-13_raw.csv.gz](https://github.com/user-attachments/files/17458761/2024-10-21_07-48-13_raw.csv.gz)
[2024-10-21_07-48-13_rowingData.csv](https://github.com/user-attachments/files/17458762/2024-10-21_07-48-13_rowingData.csv)

Whereas, with the drag set to 2 OPM did not count any strokes. See this data:
[2024-10-21_07-49-18_raw.csv.gz](https://github.com/user-attachments/files/17458771/2024-10-21_07-49-18_raw.csv.gz)
[2024-10-21_07-49-18_rowingData.csv](https://github.com/user-attachments/files/17458772/2024-10-21_07-49-18_rowingData.csv)

I plotted each of these raw files on the same graph (with time adjusted x axis). The one on top is the drag set to 10 (the strokes are detected), the one on the bottom is the drag set to 2 (no strokes are detected). With the right settings, should OPM be able to detect the drag and detect strokes appropriately?

![Untitled](https://github.com/user-attachments/assets/090a5814-9bb2-4ef9-80cd-58b6e3ca1ba5)

What do you think I should check next?

**Reply by Abasz** [reply]
These are very nice and clean curves, orm should have no issue detecting strokes in either break setting. 

I think the issue is that in case of the different break settings the recovery slope is completely different. I suppose you may need to change the `minimumRecoverySlope: 0.001,` as in case of break settings 2 this is not reached. 

From the docs:
minumumRecoverySlope can be set to 0, where OpenRowingMonitor will use a quite robust selection on an accelerating or decelerating flywheel. This is recomended as a starting point for getting stroke detection to work. It can be further optimised later (see the later section on advanced stroke detection);

**Reply by JaapvanEkris** [reply]
Could indeed very well be the minimumRecoverySlope that is too strict here. As it won't enter a recovery, it will not calculate the correct recoveryslope either, permanently blocking stroke detection.

As it is set to autoadjust, setting it to 0 initially wouldn't hurt. Even disabling autoadjustrecoveryslope wouldn't matter that much.

**Reply by alistairdickie** [reply]
Thanks for your support. This got it working at the minimum brake level on the rower (2). Setting mininiumRecoverySlope to 0 was not enough though. I had to disable autoAdjustRecoverySlope as well. 

Note the auto calculated drag factor looks a bit better too. 
Brake level 10- around 650 to 660
Brake level 6 - around 570 to 580
Brake level 2 - around 240 to 270

Now I guess I need to do a lot of rowing to tweak the settings to be like the PM5 I am used to.

**Reply by JaapvanEkris** [reply]
> Now I guess I need to do a lot of rowing to tweak the settings to be like the PM5 I am used to.

Any update on this? I'm especially interested in the flywheel inertia. I've updated the rest of the config to work with the upcoming 0.9.7, and I'd like to get this as perfect as possible.
